src/libcommon/utility/types.h

```diff
@@ -364,10 +366,21 @@ struct ItemType {
 
 enum class AppStateKey {
     // Adding a new key here requires to add it in insertDefaultAppState in parmsdbappstate.cpp
-    LastServerSelfRestart,
-    LastClientSelfRestart,
-
-    Unknown,  //!\ keep in last position (For tests) /!\\ Used for initialization, will throw error if used.
+    LastServerSelfRestartDate,
+    LastClientSelfRestartDate,
+    LastSuccessfulLogUploadeDate,
+    LastLogUploadArchivePath,
+    /* Log upload status is a string with the first character being the status and the rest being the progress percentage
+     * "A" means "Archiving" (ie. A80 means 80% of the archiving is done)
+     * "U" means "Uploading" (ie. U80 means 80% of the uploading is done)
+     * "S" means "Success" (ie. S means the archiving and upload where successful)
+     * "F" means "Failed"
+     * "C0" means "Cancel requested"
+     * "C1" means "Canceled"
+     * "N" means "None" (ie. N means we never tried to upload the logs)
+     */
+    LogUploadStatus,
```

Comment:
Is it really necessary to save the progress of the upload, archiving, ... ?
If upload fail, we will restart it from 0 anyway?

Reply:
It is used when the client closes the upload window and reopens it later during the transfer. Without this feature, the client would have to wait for the next status update to see an accurate percentage, which could take several dozen seconds if there is a large amount of log data or a poor connection.